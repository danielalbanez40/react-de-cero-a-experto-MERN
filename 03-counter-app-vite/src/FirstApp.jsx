import React from 'react';

const FirstApp = () => {
  return (
    <>
      <h1>FirstApp</h1>
      <h2>Soy un subt√≠tulo</h2>
    </>
  );
};

export default FirstApp;
