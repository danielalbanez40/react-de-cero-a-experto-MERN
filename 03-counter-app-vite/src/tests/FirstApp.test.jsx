import { render } from '@testing-library/react';
import FirstApp from '../FirstApp';

describe('Tests in <FirstApp/>', () => {
  test('should make with snapshot', () => {
    const title = 'Hola, soy Goku';
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('should show title in <h1> tag', () => {
    const title = 'Hola, soy Goku';
    const { container, getByText } = render(<FirstApp title={title} />);
    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector('h1');
    // expect(h1.innerHTML).toBe(title);
    expect(h1.innerHTML).toContain(title);

  });
});
