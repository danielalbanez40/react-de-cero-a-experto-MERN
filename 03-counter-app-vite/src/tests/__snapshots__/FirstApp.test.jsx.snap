// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp/> should make with snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Daniel Albanez
  </p>
</div>
`;
